syncTextures.js

```diff
@@ -0,0 +1,229 @@
+/** 
+ * syncs textures from the server to local development environment
+ */
+
+const fs = require('fs');
+const path = require('path');
+const https = require('https');
+const lodash = require('lodash/array');
+const axios = require('axios'); 
+// url to server-side script with updated image list (json)
+const textureList = 'https://www.project-atlantis.com/wp-content/threejs-simulation/textures/getTextures.php';
+// url to location of remote texture files
+const textureSource = 'https://www.project-atlantis.com/wp-content/threejs-simulation/textures';
+// local texture folder
+const textureFolder = './textures';
+// delay (milliseconds) between file download requests --- server overload prevention
+const delay = '10';
+
+/**
+ * scans local textures
+ *
+ * @param {string}  localPath   // path to local textures
+ * @param {array}   fileArray   // callback to merge textures into single array
+ *
+ * @return {array}  fileArray   // list of local textures
+ */
+function getLocalTextures(localPath, fileArray) {
+  files = fs.readdirSync(localPath);
+  fileArray = fileArray || []
+
+  files.forEach(function(file) {
+    if (fs.statSync(localPath + "/" + file).isDirectory()) {
+      fileArray = getLocalTextures(localPath + "/" + file, fileArray)
+    } else {
+      fileArray.push(path.join(localPath, "/", file))
+    }
+  })
+
+  fileArray.forEach((element, index) => {
+    fileArray[index] = element.replace(/^textures/, "");
+  });
+
+  return fileArray;
+}
+
+/**
+ * creates any missing texture directories 
+ *
+ * @param {array} dirList   // list of directories to verify
+ */
+function verifyTextureFolders(dirList) {
+  try {
+    if (!fs.existsSync(textureFolder)) {
+      fs.mkdirSync(textureFolder);
+      console.log('Created ' + textureFolder);
+    }
+  } catch (err) {
+    console.error(err);
+  }
+  
+  dirList.forEach( (dir) => {
+    dir = path.join(__dirname, textureFolder, dir);
+    try {
+      if(!fs.existsSync(dir)) {
+        fs.mkdirSync(dir, { recursive: true });
+        console.log('Created ' + dir);
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  })
+}
+
+/**
+ * parses the list of remote textures to determine directory structure to create locally
+ *
+ * @param {array}  remoteList               // array with list of remote textures
+ *
+ * @return {Promise<array>} directoryTree   // array containing folders to verify exist
+ */
+async function getRemoteDirectoryTree() {
+  let remoteList = await getRemoteTextures(textureList);
+  let directoryTree = [];
+  let dir;
+  if (remoteList) {
+    remoteList.forEach( (texture) => {
+      dir = texture.replace(/[^/]*$/, '');
+      directoryTree.push(dir);
+    });
+    directoryTree = [... new Set(directoryTree)];
+    return directoryTree;
+  }
+}
+
+/**
+ * pulls list (json) of remote textures from the server
```

Comment:
A more explicate description would help other people to understand the code. Something like...
```suggestion
 * if the url provided is valid, this function returns another function that interfaces with a server-side php script that ... and ... uses it to create a list of <file types> that hosted on the project-atlantis.com webserver. 
```